generators/pkg/manager/manage_test.go

```diff
@@ -0,0 +1,205 @@
+package manager
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+type testClient struct {
+	client.Writer
+	list []runtime.Object
+}
+
+func (t *testClient) Create(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error {
+	t.list = append(t.list, obj)
+	return nil
+}
+
+func (t *testClient) Delete(ctx context.Context, obj runtime.Object, opts ...client.DeleteOption) error {
+	t.list = append(t.list, obj)
+	return nil
+}
+
+// testdata
+var resources = []*unstructured.Unstructured{
+	{
+		Object: map[string]interface{}{
+			"name":       "test-task",
+			"kind":       "Task",
+			"apiVersion": "tekton.dev/v1beta1",
+		},
+	},
+
+	{
+		Object: map[string]interface{}{
+			"name":       "test-task",
+			"kind":       "Task",
+			"apiVersion": "tekton.dev/v1beta1",
+			"metadata": map[string]interface{}{
+				"namespace": "default",
+			},
+		},
+	},
+
+	{
+		Object: map[string]interface{}{
+			"name":       "test-task",
+			"kind":       "Task",
+			"apiVersion": "tekton.dev/v1beta1",
+			"metadata": map[string]interface{}{
+				"namespace": "default",
+			},
+		},
+	},
+
+	{
+		Object: map[string]interface{}{
+			"apiVersion": "tekton.dev/v1beta1",
+			"kind":       "Task",
+			"metadata": map[string]interface{}{
+				"namespace": "default",
+				"name":      "echo-hello-world",
+			},
+			"spec": map[string]interface{}{
+				"steps": map[string]interface{}{
+					"name":    "echo",
+					"image":   "ubuntu",
+					"command": []interface{}{string("echo")},
+					"args":    []interface{}{string("Hello world")},
+				},
+			},
+		},
+	},
+}
+
+func TestApplyResource(t *testing.T) {
+	tables := []struct {
+		u    *unstructured.Unstructured
+		want []runtime.Object
+	}{
+		{
+			resources[0],
+			[]runtime.Object{
+				&unstructured.Unstructured{
+					Object: map[string]interface{}{
+						"name":       "test-task",
+						"kind":       "Task",
+						"apiVersion": "tekton.dev/v1beta1",
+						"metadata": map[string]interface{}{
+							"namespace": "default",
+						},
+					},
+				},
+			},
+		},
+
+		{
+			resources[1],
+			[]runtime.Object{
+				resources[1],
+			},
+		},
+
+		{
+			resources[2],
+			[]runtime.Object{
+				resources[2],
+			},
+		},
+
+		{
+			resources[3],
+			[]runtime.Object{
+				resources[3],
+			},
+		},
+	}
+
+	for _, table := range tables {
+		b, err := yaml.Marshal(table.u)
```

Comment:
With table driven tests like these, we should use subtests i.e a `t.Run()` for each case in the table.
See https://ieftimov.com/post/testing-in-go-subtests/

Reply:
Cool. I'll take a look and change my code.